pkg/services/types_test.go

```diff
@@ -85,18 +86,117 @@ func Test_ValidateOrderBy(t *testing.T) {
 		},
 	}
 
-	RegisterTestingT(t)
-
+	g := NewWithT(t)
 	for _, testcase := range tests {
 		tt := testcase
 		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
 			la := NewListArguments(tt.params)
 			err := la.Validate(tt.validParams)
-			if tt.wantErr {
-				Expect(err).To(HaveOccurred())
-			} else {
-				Expect(err).NotTo(HaveOccurred())
+			g.Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+func Test_NewListArguments(t *testing.T) {
+	page := "5"
+	size := "-1"
+	search := "search"
+	overriddenListArgs := &ListArguments{
+		Page:   5,
+		Size:   65500,
+		Search: "search",
+	}
+	type args struct {
+		params url.Values
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want *ListArguments
+	}{
+		{
+			name: "should override default param",
+			args: args{
+				params: url.Values{
+					"page":   []string{page},
+					"size":   []string{size},
+					"search": []string{search},
+				},
+			},
+			want: overriddenListArgs,
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			g.Expect(NewListArguments(tt.args.params)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestListArguments_Validate(t *testing.T) {
+	type fields struct {
+		Page     int
+		Size     int
+		Preloads []string
+		Search   string
+		OrderBy  []string
+	}
+	type args struct {
+		acceptedOrderByParams []string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "should return an error if page is < 0",
+			fields: fields{
+				Page: -1,
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+		},
+		{
+			name: "should return an error if page is < 1",
+			fields: fields{
+				Page: 0,
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+		},
+		{
+			name: "should return an error if there are too many keywords in orderBy",
+			fields: fields{
+				Page:    1,
+				Size:    100,
+				OrderBy: []string{"region desc name owner"},
+			},
+			args: args{
+				acceptedOrderByParams: getValidTestParams(),
+			},
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			la := &ListArguments{
+				Page:     tt.fields.Page,
+				Size:     tt.fields.Size,
+				Preloads: tt.fields.Preloads,
+				Search:   tt.fields.Search,
+				OrderBy:  tt.fields.OrderBy,
 			}
+			err := la.Validate(tt.args.acceptedOrderByParams)
+			g.Expect(err).To(HaveOccurred())
```

Comment:
All the tests here are validating that errors are detected. Can we add a test to check that valid parameters are correctly validated?

Reply:
I believe that the correct parameters are being tested when the function is being called here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1129/files#diff-68ad7f79ef378bd62790c95989296d431e4221a4db8601f778b0cf5ca83d7214R95 .It was just these three errors where missing from the coverage.

Reply:
The test you are referring to is validating the order by arguments and, incidentally, validates the other args as well.

Each test should be auto-contained, so if this testing if argument validation works, it should test that both valid and invalid arguments are correctly evaluated, without relying on other tests.

The other test could change and it could stop having the side effect of validating other args than order by in the future.